aten/src/ATen/native/Loss.cpp

```diff
@@ -314,53 +315,101 @@ Tensor& binary_cross_entropy_backward_out_cpu(const Tensor& grad, const Tensor&
     return grad_input;
 }
 
-Tensor binary_cross_entropy_with_logits(const Tensor& input, const Tensor& target, const c10::optional<Tensor>& weight_opt, const c10::optional<Tensor>& pos_weight_opt, int64_t reduction) {
+Tensor binary_cross_entropy_with_logits(
+    const Tensor& input,
+    const Tensor& target,
+    const c10::optional<Tensor>& weight_opt,
+    const c10::optional<Tensor>& pos_weight_opt,
+    int64_t reduction) {
   // See [Note: hacky wrapper removal for optional tensor]
-  c10::MaybeOwned<Tensor> weight_maybe_owned = at::borrow_from_optional_tensor(weight_opt);
+  c10::MaybeOwned<Tensor> weight_maybe_owned =
+      at::borrow_from_optional_tensor(weight_opt);
   const Tensor& weight = *weight_maybe_owned;
-  const Tensor& pos_weight = c10::value_or_else(pos_weight_opt, [] {return Tensor();});
-
-    Tensor loss;
-    auto max_val = (-input).clamp_min_(0);
-    if (pos_weight.defined()) {
-        // pos_weight need to be broadcasted, thus mul(target) is not inplace.
-        auto log_weight = (pos_weight - 1).mul(target).add_(1);
-        loss = (1 - target).mul_(input).add_(log_weight.mul_(((-max_val).exp_().add_((-input - max_val).exp_())).log_().add_(max_val)));
+  const Tensor& pos_weight =
+      c10::value_or_else(pos_weight_opt, [] { return Tensor(); });
+
+  Tensor loss;
+  auto max_val = (-input).clamp_min_(0);
+  auto hasSubclassTensors = at::areAnyTensorSubclassLike({target, input});
```

Comment:
Shouldn't this also check if weight and pos_weight are subclass?

Reply:
I thought the pos_weight doesn't need to be checked because: `auto log_weight = (pos_weight - 1).mul(target).add_(1);` is the only place where it could be mutated and the first mul is out of place so it wouldn't matter.  I check whether the weight tensor is subclassed only if it is defined.. I thought that was enough ?

Reply:
The code as written should be sufficient